crates/bevy_render/src/texture/dds.rs

```diff
@@ -45,7 +48,18 @@ pub fn dds_buffer_to_image(
         depth_or_array_layers,
     }
     .physical_size(texture_format);
-    image.texture_descriptor.mip_level_count = dds.get_num_mipmap_levels();
+    let mip_map_level = match dds.get_num_mipmap_levels() {
+        0 => {
+            #[cfg(debug_assertions)]
+            warn_once!(
+                "Mipmap levels for texture {} are 0, bumping them to 1",
+                name
+            );
+            1
+        }
+        t => t,
+    };
+    image.texture_descriptor.mip_level_count = mip_map_level;
```

Comment:
I feel like I would have just done get_num_mipmap_levels().max(1) and added a comment that some older dds files may return that they have 0 mip map levels. Then this PR is a one line change.

Adding the name stuff seems useful, but separate.

Reply:
This was my initial change, and other reviewers started to request more and more and we ended up here.

Reply:
Ok, fair enough.